orderbook/src/maintenance.rs

```diff
@@ -0,0 +1,42 @@
+use crate::database::Database;
+use crate::event_updater::EventUpdater;
+use crate::orderbook::Orderbook;
+use anyhow::Result;
+use futures::future::join_all;
+use shared::current_block::Maintaining;
+use std::sync::Arc;
+
+/// Collects all service components requiring maintenance on each new block
+pub struct ServiceMaintenance {
+    storage: Arc<Orderbook>,
+    database: Database,
+    event_updater: EventUpdater,
+}
+
+impl ServiceMaintenance {
+    pub fn new(storage: Arc<Orderbook>, database: Database, event_updater: EventUpdater) -> Self {
```

Comment:
Should this be agnostic to the number and kind of components it takes and just work on `impl Iterator<Item = Arc<dyn Maintaining>>`?

Reply:
The original motivation was that we weren't sure whether or not there were ordering requirements (see discussion here: https://github.com/gnosis/gp-v2-services/pull/616#discussion_r635212228). Now that we determined that there aren't any - I agree that act on a collection of `Arc<dyn Maintaining>`s.

We could then even move this into the `shared` crate.

Reply:
Will do. Although we may find in the future that they can't necessarily all be run in parallel.

Reply:
With this `impl Iterator` I get the warning that ``impl Trait` not allowed outside of function and inherent method return types [E0562]`. I suspect I will have to go with `Vec<Arc<dyn Maintaining>>` if thats alright with everyone.

Reply:
I believe at least on the struct you need to chose a concrete container type (or make the struct templated, but that's overkill). But Vec everywhere works as well.